csharp/IceStorm/Weather/Station/Program.cs

```diff
@@ -38,14 +40,15 @@
 // The proxy returned by createAsync and retrieveAsync is never null.
 Debug.Assert(topic is not null);
 
-// Register our weather station with the topic.
-// subscribeAndGetPublisherAsync returns a publisher proxy that we don't need here.
-_ = await topic.subscribeAndGetPublisherAsync(theQoS: [], weatherStation);
-
 // Start dispatching requests.
 adapter.activate();
 Console.WriteLine("Listening...");
 
+// Register our weather station with the topic.
```

Comment:
Moved the call to subscribeAndGetPublisher _after_ activate. 

If activate takes a while (very unlikely here, most likely with IceGrid), we don't want IceStorm to fail to establish a connection to Station while activate is in progress. So best model is to subscribe after activate.